config.go

```diff
@@ -16,6 +16,12 @@ type Config struct {
 	// Packages is the slice of Nix packages that devbox makes available in
 	// its environment.
 	Packages []string `cue:"[...string]" json:"packages"`
+
+	// Shell configures the devbox shell environment.
+	Shell struct {
+		// InitHook contains commands that will run at shell startup.
+		InitHook string `json:"init_hook,omitempty"`
```

Comment:
A few quick questions about this format: 
* Are there more non-hook fields that we foresee adding to the `shell` struct? 
* Should `stages` be in a `Docker` struct or something else, so that it's clearer they only affect Docker and hooks only affect shells?

Reply:
Not a fan of the word `Docker` or `Container` in general. Feels like we are exposing implementation details. Some fields are for `shell`, others for `build`. At least that's how I've been thinking about them.

Reply:
For shell, we discussed possibly letting users pick which shell runs in devbox. I was thinking it might become something like:

```json
{
  "packages": ["go"],
  "shell": {
    "type": "zsh",
    "init_hook": "echo hello from a hook!"
  }
}
```

I'm not sure what we'd group the other fields under. They seem general enough that I think they're fine where they are. No strong opinion on that though.